File: angular-casestudy/src/app/customer.ts
export class Customer {
id: number;
name: string;
gender: number;
idCarrd: string;
phone: string;
customertype: string;

  constructor(id: number, name: string, gender: number, idCarrd: string, phone: string, customertype: string) {
    this.id = id;
    this.name = name;
    this.gender = gender;
    this.idCarrd = idCarrd;
    this.phone = phone;
    this.customertype = customertype;
  }
}

File: demo1/src/main/java/demo1/model/Customer.java
package demo1.model;

import javax.persistence.*;
import java.util.Set;

@Entity
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String customerName;
    private String birthday;
    private int gender;
    private String idCard;
    private String phone;
    private String email;
    private String address;
    @ManyToOne
    @JoinColumn(name = "customer_type_id", referencedColumnName = "customerTypeId")
    private CustomerType customerType;
}
